lib/std/sort.zig

```diff
@@ -89,23 +93,30 @@ pub fn heapContext(a: usize, b: usize, context: anytype) void {
     }
 }
 
-fn siftDown(a: usize, root: usize, n: usize, context: anytype) void {
-    var node = root;
-    while (true) {
-        var child = a + 2 * (node - a) + 1;
-        if (child >= n) break;
+fn siftDown(a: usize, cur: usize, b: usize, context: anytype) void {
+    var child = (math.mul(usize, cur - a, 2) catch return) + a + 1;
```

Comment:
What exactly happens when this multiplication does overflow? Is there a case where we silently fail to sort fully? It's probably worth a comment in any case.

Reply:
I think I see, does an overflow here mean that the recursion should terminate as we are looking past the end of the heap?

Reply:
style: Comments above a block of statements. If you feel like expressing an invariant, then it belongs below the statement (and personally I use `invariant:` but thats up to you).

Reply:
Yes, an overflow here means we are done sifting that particular item. It is overshooting the boundary, exactly the same thing we check for in the next line of code. I thought it was clear by the comments underneath, but I suppose I could add another comment.